projects/components/src/checkbox/checkbox.component.ts

```diff
@@ -8,31 +9,66 @@ import { MatCheckboxChange } from '@angular/material/checkbox';
   template: `
     <mat-checkbox
       labelPosition="after"
-      [checked]="this.checked"
-      [disabled]="this.disabled"
-      (change)="onCheckboxChange($event)"
+      [checked]="this.isChecked"
+      [disabled]="this.isDisabled"
+      (change)="this.onCheckboxChange($event)"
       class="ht-checkbox"
-      [ngClass]="{ disabled: this.disabled }"
+      [ngClass]="{ disabled: this.isDisabled }"
     >
       <ht-label class="label" *ngIf="this.label !== undefined && this.label !== ''" [label]="this.label"></ht-label>
     </mat-checkbox>
-  `
+  `,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: CheckboxComponent,
+      multi: true
+    }
+  ]
 })
-export class CheckboxComponent {
+export class CheckboxComponent implements ControlValueAccessor {
   @Input()
   public label?: string;
 
   @Input()
-  public checked: boolean | undefined;
+  public set checked(checked: boolean | undefined) {
+    this.isChecked = checked ?? false;
+  }
 
   @Input()
-  public disabled: boolean | undefined;
+  public set disabled(disabled: boolean | undefined) {
```

Comment:
why do we need these setters?

Reply:
To avoid mutating the input property.

Reply:
I am not sure I follow.  Can you please provide more info or give example?